he_twitter_miner/src/twitterMiner.py

```diff
@@ -70,32 +41,113 @@ def insert_data(self, tweet):
                         ).format(self.processed_count, hashtag)
                     )
                 self.processed_count += 1
-
-        elif 'warning' in tweet:
-            logging.warning(tweet)
-
         return True
 
-    def on_disconnect(self, notice):
-        logging.error(notice)
-
 
 class StdOutListener(StreamListener):
 
-    def __init__(self):
-        self.data_inserter = DataInserter()
-
-    def on_data(self, data):
-        self.data_inserter.insert_data(data)
-        return True
-
-    def on_error(self, status):
-        logging.info(status)
+    def __init__(self, hashtag_exchange_collection):
+        self.data_inserter = DataInserter(hashtag_exchange_collection)
+        super(StdOutListener, self).__init__()
+      
+    def on_data(self, raw_data):
+        data = json.loads(raw_data)
+        if 'in_reply_to_status_id' in data:
+            return self.data_inserter.insert_data(data)
+        return super(StdOutListener, self).on_data(raw_data)
+  
+    def keep_alive(self):
+        logging.info("kept alive message received")
+        return
+
+    def on_status(self, status):
+        '''handled on on_data instead'''
+        return
+
+    def on_exception(self, exception):
+        logging.info("exception received notice: {}".format(exception))
+        return False
+
+    def on_event(self, status):
+        logging.info("event received: {}".format(status))
+        """Called when a new event arrives we dont care about these messages"""
+        return
+
+    def on_direct_message(self, status):
+        """Called when a new direct message arrives we dont care about these messages"""
+        return
+
+    def on_friends(self, friends):
+        """Called when a friends list arrives.
+        friends is a list that contains user_id
+        """
+        return
+
+    def on_limit(self, track):
+        """Called when a limitation notice arrives"""
+        return
```

Comment:
ditto - in empty functions, just use `pass` (although not much difference in this case)